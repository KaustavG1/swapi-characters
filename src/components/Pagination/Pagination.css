.pagination-container {
  display: flex;
  gap: 5px;
  justify-content: center;
}

.pagination-buttons {
  font-size: 1.5rem;
}

.pagination-button-disabled {
  cursor: not-allowed !important;
}
