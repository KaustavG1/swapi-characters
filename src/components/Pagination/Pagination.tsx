import Button from "../common/Button/Button";
import "./Pagination.css";

function Pagination() {
  return (
    <div className="pagination-container">
      <Button text="<" disabled={false} />
      <Button text=">" disabled={false} />
    </div>
  );
}

export default Pagination;
