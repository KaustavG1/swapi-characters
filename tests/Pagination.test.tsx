import { it, expect, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import Pagination from "../src/components/Pagination/Pagination";

describe("Pagination component", () => {
  it("renders 'Pagination' component with disbled previous", () => {
    const onPreviousClick = vi.fn();
    const onNextClick = vi.fn();
    const { container } = render(
      <Pagination
        disablePrev={true}
        disableNext={false}
        onPreviousClick={onPreviousClick}
        onNextClick={onNextClick}
      />
    );
    const paginationPrevious = screen.getByText(/</i);
    expect(paginationPrevious).toBeInTheDocument();
    expect(paginationPrevious.parentNode).toBeDisabled();
    const paginationNext = screen.getByText(/>/i);
    expect(paginationNext).toBeInTheDocument();
    expect(paginationNext.parentNode).not.toBeDisabled();
    fireEvent.click(paginationNext);
    expect(onNextClick).toHaveBeenCalledOnce();
    expect(container).toMatchSnapshot();
  });

  it("renders 'Pagination' component with disbled next", () => {
    const onPreviousClick = vi.fn();
    const onNextClick = vi.fn();
    const { container } = render(
      <Pagination
        disablePrev={false}
        disableNext={true}
        onPreviousClick={onPreviousClick}
        onNextClick={onNextClick}
      />
    );
    const paginationPrevious = screen.getByText(/</i);
    expect(paginationPrevious).toBeInTheDocument();
    expect(paginationPrevious.parentNode).not.toBeDisabled();
    const paginationNext = screen.getByText(/>/i);
    expect(paginationNext).toBeInTheDocument();
    expect(paginationNext.parentNode).toBeDisabled();
    fireEvent.click(paginationPrevious);
    expect(onPreviousClick).toHaveBeenCalledOnce();
    expect(container).toMatchSnapshot();
  });
});
