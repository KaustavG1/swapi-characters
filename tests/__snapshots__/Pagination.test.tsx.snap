// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination component > renders 'Pagination' component with disbled next 1`] = `
<div>
  <div
    class="pagination-container"
  >
    <div>
      <button
        class="pagination-buttons false button-default"
      >
        <span />
        <span>
          &lt;
        </span>
      </button>
    </div>
    <div>
      <button
        class="pagination-buttons pagination-button-disabled button-default"
        disabled=""
      >
        <span />
        <span>
          &gt;
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Pagination component > renders 'Pagination' component with disbled previous 1`] = `
<div>
  <div
    class="pagination-container"
  >
    <div>
      <button
        class="pagination-buttons pagination-button-disabled button-default"
        disabled=""
      >
        <span />
        <span>
          &lt;
        </span>
      </button>
    </div>
    <div>
      <button
        class="pagination-buttons false button-default"
      >
        <span />
        <span>
          &gt;
        </span>
      </button>
    </div>
  </div>
</div>
`;
